app/controllers/api/v1/basic_auth_api/protocol_subscriptions_controller.rb

```diff
@@ -49,13 +49,17 @@ def set_protocol_subscription
         def start_date
           return Time.zone.now if protocol_subscription_create_params[:start_date].blank?
 
-          Time.zone.parse(protocol_subscription_create_params[:start_date])
+          # The start date cannot be in the past.
+          [Time.zone.parse(protocol_subscription_create_params[:start_date]), Time.zone.now].max
         end
 
         def end_date
           return nil if protocol_subscription_create_params[:end_date].blank?
 
-          Time.zone.parse(protocol_subscription_create_params[:end_date])
+          # The duration between start and end date should be at least one hour.
```

Comment:
why is this? Can you add a comment why it should be at least 1 hour?

Reply:
So people have at least one hour to fill out the measurement.

Reply:
I added a comment.